engine/src/main/scala/cromwell/engine/EngineELF.scala

```diff
@@ -2,21 +2,31 @@ package cromwell.engine
 
 import java.nio.file.{FileSystem, FileSystems}
 
+import com.typesafe.config.ConfigFactory
 import cromwell.backend.wdl.{PureFunctions, ReadLikeFunctions}
 import cromwell.core.WorkflowOptions
+import cromwell.engine.backend.EnhancedWorkflowOptions._
+import cromwell.filesystems.gcs.{GcsFileSystem, GcsFileSystemProvider, GoogleConfiguration}
+import lenthall.config.ScalaConfig._
+import spray.json.JsObject
 import wdl4s.expression.WdlStandardLibraryFunctions
 import wdl4s.values.{WdlFile, WdlValue}
 
 import scala.util.{Failure, Try}
 
 class EngineELF(workflowOptions: WorkflowOptions) extends WdlStandardLibraryFunctions with ReadLikeFunctions with PureFunctions {
+  def gcsFileSystem: Option[GcsFileSystem] = {
+    for {
+      authModeString <- ConfigFactory.load.getStringOption("engine.filesystems.gcs.auth")
+      authMode <- GoogleConfiguration.Instance.auth(authModeString).toOption
```

Comment:
Oh yeah, I agree that we should not allow *invalid* config... sorry I misunderstood.  Like if they specify `application-foobar` then yeah, some kind of exception should be thrown... let me see what I can do about that.